server/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -1092,4 +1122,21 @@ public void testNonSecureSettingInKeystore() {
         assertThat(e.getMessage(), containsString("must be stored inside elasticsearch.yml"));
     }
 
+    public void testLogSettingUpdate() throws Exception {
+        IndexMetaData metaData = newIndexMeta("index1",
+            Settings.builder().put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), "20s").build());
+        IndexSettings settings = new IndexSettings(metaData, Settings.EMPTY);
+        Logger logger = settings.getScopedSettings().getLogger();
+
+        MockAppender appender = new MockAppender("trace_appender");
```

Comment:
We already have some utility classes for asserting things about logging. I suggest this test:

```

    @TestLogging(value="org.elasticsearch.common.settings.IndexScopedSettings:INFO",
        reason="to ensure we log INFO-level messages from IndexScopedSettings")
    public void testLogSettingUpdate() throws Exception {
        final IndexMetaData metaData = newIndexMeta("index1",
            Settings.builder().put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), "20s").build());
        final IndexSettings settings = new IndexSettings(metaData, Settings.EMPTY);

        final MockLogAppender mockLogAppender = new MockLogAppender();
        mockLogAppender.addExpectation(new MockLogAppender.SeenEventExpectation(
            "message",
            "org.elasticsearch.common.settings.IndexScopedSettings",
            Level.INFO,
            "updating [index.refresh_interval] from [20s] to [10s]") {
            @Override
            public boolean innerMatch(LogEvent event) {
                return event.getMarker().getName().equals(" [index1]");
            }
        });
        mockLogAppender.start();
        final Logger logger = LogManager.getLogger(IndexScopedSettings.class);
        try {
            Loggers.addAppender(logger, mockLogAppender);
            settings.updateIndexMetaData(newIndexMeta("index1",
                Settings.builder().put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), "10s").build()));

            mockLogAppender.assertAllExpectationsMatched();
        } finally {
            Loggers.removeAppender(logger, mockLogAppender);
            mockLogAppender.stop();
        }
    }
```

We need the `@TestLogging` annotation since it's possible to adjust logging settings for the whole test task in a way that would make this fail otherwise (e.g. setting `logger.org.elasticsearch: WARN`)

Also we can assert this about `LogManager.getLogger(IndexScopedSettings.class)`, there's no need for a public method that exposes the logger on the `AbstractScopedSettings`.

Also it's important to clean things up in a `finally` block whether the test passed or failed.